File: Algorithms/InvertBinaryTree/InvertBinaryTree_iter.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (!root) return NULL;
        std::queue<TreeNode*> q;
        
        q.push(root);
        while (q.size() > 0){
            TreeNode* ptr = q.front();
            q.pop();
            std::swap(ptr->left, ptr->right);
            if (ptr->left) q.push(ptr->left);
            if (ptr->right) q.push(ptr->right);
        }
        return root;
    }
};

File: Algorithms/InvertBinaryTree/InvertBinaryTree_iter.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root: return None
        q = deque([root])
        while q:
            ptr = q.pop()
            ptr.left, ptr.right = ptr.right, ptr.left
            if ptr.left: q.appendleft(ptr.left)
            if ptr.right: q.appendleft(ptr.right)
        return root